### ISSUE TITLE
Vertical size of FlowBox is measured for vertical layout instead of the flow layout used

### ISSUE THREAD

**Issue by nicoulaj** [flex-users/flexlib#68]
Originally filed by ali.rantakari on 2007-12-18T15:47:19

What steps will reproduce the problem?
1. Add a new FlowBox
2. Set the vertical size of the FlowBox to 100%
3. Add stuff inside the FlowBox

What is the expected output? What do you see instead?
Stuff inside FlowBox is laid out as expected, in a flow layout. Expected to see vertical size of the 
FlowBox adjust to this layout but instead it remains the size it would be if its children would be 
laid according to a vertical layout (i.e. 'stacked' on top of each other)

What version of the product are you using? On what operating system?
flexlib-.2.1, Flash Player 9,0,47,0 on Mac OS 10.5.1

Please provide any additional information below.
This behavior is prevalent also in the examples for the FlowBox container

**Comment by nicoulaj** [comment]
Updated by ali.rantakari on 2007-12-18T15:48:47

Typo: Step two (2) should say "width" instead of "vertical size"

**Comment by nicoulaj** [comment]
Updated by dmcc...@gmail.com on 2007-12-25T14:36:56

Added label FlowBox

**Comment by nicoulaj** [comment]
Updated by dmcc...@gmail.com on 2007-12-25T22:13:29

Original ticket set owner to dmccune
Original ticket set status to Started (we converted to open)

**Comment by nicoulaj** [comment]
Updated by dmcc...@gmail.com on 2007-12-26T11:11:30

Added label Milestone-Release.2.2

**Comment by nicoulaj** [comment]
Updated by dmcc...@gmail.com on 2007-12-26T11:33:13

Removed label FlowBox
Added label FlexLib-FlowBox

**Comment by nicoulaj** [comment]
Updated by dmcc...@gmail.com on 2008-01-03T08:30:11

Original ticket set status to Fixed (we converted to closed)

**Comment by nicoulaj** [comment]
Updated by ali.rantakari on 2008-01-22T14:08:16

This issue is still occurring in the FlowBox examples, is it really fixed?

**Comment by nicoulaj** [comment]
Updated by dmcc...@gmail.com on 2008-01-25T06:33:05

The FlowBox example was a bit misleading, as it had the FlowBoxes contained in Panels
that had both height and width set at 250 pixels. So it appeared as if the FlowBox
containers were taking up more height than they really were. I'm updating the sample
to show the true height of the container when it measures itself.

**Comment by nicoulaj** [comment]
Updated by ali.rantakari on 2008-05-23T11:22:06

This problem is still occurring if the container width is set to 100% (see attached screenshot.)

**Comment by nicoulaj** [comment]
Updated by cshreff...@gmail.com on 2008-06-17T21:15:42

I too am seeing this behavior.  This does not appear to be fixed.

**Comment by nicoulaj** [comment]
Updated by bensme...@gmail.com on 2008-07-07T14:13:52

Jup, same here.

**Comment by nicoulaj** [comment]
Updated by feedw...@gmail.com on 2008-07-17T17:59:26

I was able to fix this problem by incorporating a bit of code I borrowed from the
Text control.  Take a look at the standard Text.as control and everywhere you see
"isSpecialCase()", copy over the code.  Wasn't too hard.

**Comment by nicoulaj** [comment]
Updated by paulbe...@gmail.com on 2008-07-23T19:54:46

can you post the fix

**Comment by nicoulaj** [comment]
Updated by feedw...@gmail.com on 2008-07-23T20:57:01

These changes are only lightly tested.  Use at your own risk.  

All in flexlib/containers/utilityClasses/FlowLayout.as:
- add this function:
  
  private function isMeasureSpecialCase():Boolean
  {
      var left:Number = target.getStyle("left");
      var right:Number = target.getStyle("right");
  
  ```
  return (!isNaN(target.percentWidth) || (!isNaN(left) && !isNaN(right))) &&
         isNaN(target.explicitHeight) &&
         isNaN(target.percentHeight);
  ```
  
  }
- at the end of measure(), add this code:
  
  ```
  if (isMeasureSpecialCase())
  {
      if (!isNaN(lastUnscaledWidth))
      {
          doLayout(lastUnscaledWidth, false);
      }
      else
      {
          // We're not ready to measure yet.
          // We need updateDisplayList() to first tell us the unscaledWidth
          // that has been calculated from the percentWidth.
          target.measuredWidth = 0;
          target.measuredHeight = 0;
      }
      return;
  }
  ```
- at the beginning of updateDisplayList add this code:
  
  ```
  if (isMeasureSpecialCase())
  {
      var firstTime:Boolean = isNaN(lastUnscaledWidth) || lastUnscaledWidth !=
  ```
  
  unscaledWidth;
          lastUnscaledWidth = unscaledWidth;
          if (firstTime)
          {
              target.invalidateSize();
              return;
          }
      }

**Comment by nicoulaj** [comment]
Updated by paulbe...@gmail.com on 2008-07-23T22:19:27

thank you kind sir!!!!!

**Comment by nicoulaj** [comment]
Updated by herz...@gmail.com on 2009-06-10T15:59:29

Was this fixed? I'm working with version 2.4 100% height is still not ok
